File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.java
class Solution {
    public int minSteps(String s, String t) {
        Map<Character, Integer> cnt = new HashMap<>();
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            int pre = cnt.getOrDefault(s.charAt(i), 0);
            cnt.put(s.charAt(i), pre + 1);
        }
        
        for (int i = 0; i < t.length(); i++) {
            int tmp = cnt.getOrDefault(t.charAt(i), 0);
            if (tmp > 0) {
                cnt.put(t.charAt(i), tmp-1);
                res++;
            }
        }
        return t.length() - res;
    }
}

File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.py
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        cnt = Counter(t)
        res = 0
        for st in s:
            if st in cnt and cnt[st] > 0:
                cnt[st] -= 1
                res += 1
        
        return len(s) - res